tests/core/eth-module/test_transactions.py

```diff
@@ -202,6 +209,18 @@ def test_unmined_transaction_wait_for_receipt(w3, request_mocker):
         assert txn_receipt["blockHash"] is not None
 
 
+def test_eth_wait_for_transaction_receipt_transaction_indexing_in_progress(
+    w3, request_mocker
+):
+    i = _tx_indexing_response_iterator()
+    with request_mocker(
+        w3,
+        mock_responses={"eth_getTransactionReceipt": lambda *_: next(i)},
```

Comment:
This iterator approach is really cool! 😎

Reply:
yeah, this is way cleaner than the v6 way! nice!